lib/bitcoin/BitcoinProcessor.ts

```diff
@@ -0,0 +1,590 @@
+        break;
+      case 'livenet':
+        bitcoreNetwork = Networks.livenet;
+        break;
+      case 'testnet':
+        bitcoreNetwork = Networks.testnet;
+        break;
+    }
+    return new PrivateKey(undefined, bitcoreNetwork).toWIF();
+  }
+
+  /**
+   * Initializes the Bitcoin processor
+   */
+  public async initialize () {
+    console.debug('Initializing TransactionStore');
+    await this.transactionStore.initialize();
+    const lastKnownTransaction = await this.transactionStore.getLastTransaction();
+    if (lastKnownTransaction) {
+      console.info(`Last known block ${lastKnownTransaction.transactionTime} (${lastKnownTransaction.transactionTimeHash})`);
+      this.lastSeenBlock = { height: lastKnownTransaction.transactionTime, hash: lastKnownTransaction.transactionTimeHash };
+      this.lastSeenBlock = await this.processTransactions(this.lastSeenBlock);
+    } else {
+      this.lastSeenBlock = await this.processTransactions();
+    }
+    this.periodicPoll();
+  }
+
+  /**
+   * Gets the blockchain time of the given time hash.
+   * Gets the latest logical blockchain time if time hash is not given.
+   * @param hash Blockchain time hash.
+   * @returns the current or associated blockchain time of the given time hash.
+   */
+  public async time (hash?: string): Promise<IBlockchainTime> {
+    console.info(`Getting time ${hash ? 'of time hash ' + hash : ''}`);
+    let request: any;
+    if (hash) {
+      request = {
+        method: 'getblock',
+        params: [
+          hash, // hash of the block
+          true, // block details
+          false // transaction details
+        ]
+      };
+    } else {
+      const blockHeight = await this.getCurrentBlockHeight();
+      request = {
+        method: 'getblockbyheight',
+        params: [
+          blockHeight,  // height of the block
+          true, // block details
+          false // transaction details
+        ]
+      };
+    }
+    const response = await this.rpcCall(request);
+    return {
+      hash: response.hash,
+      time: response.height
+    };
+  }
+
+  /**
+   * Fetches Sidetree transactions in chronological order from since or genesis.
+   * @param since A transaction number
+   * @param hash The associated transaction time hash
+   * @returns Transactions since that blocktime
+   */
+  public async transactions (since?: number, hash?: string): Promise<{
+    moreTransactions: boolean,
+    transactions: ITransaction[]
+  }> {
+    if ((since && !hash) ||
+        (!since && hash)) {
+      throw new SidetreeError(httpStatus.BAD_REQUEST);
+    } else if (since && hash) {
+      if (!await this.verifyBlock(TransactionNumber.getBlockNumber(since), hash)) {
+        console.info('Requested transactions hash mismatched blockchain');
+        throw new SidetreeError(httpStatus.BAD_REQUEST, 'invalid_transaction_number_or_time_hash');
+      }
+    } else {
+      since = TransactionNumber.construct(this.genesisBlockNumber, 0);

```

Comment:
Probably corner case, but technically incorrect since this would exclude the transactions contained in the genesis block. I'll make an update.